airflow/triggers/base.py

```diff
@@ -137,3 +150,96 @@ def __eq__(self, other):
         if isinstance(other, TriggerEvent):
             return other.payload == self.payload
         return False
+
+    @provide_session
+    def handle_submit(self, *, task_instance: TaskInstance, session: Session = NEW_SESSION) -> None:
+        """
+        Handle the submit event for a given task instance.
+
+        This function sets the next method and next kwargs of the task instance,
+        as well as its state to scheduled. It also adds the event's payload
+        into the kwargs for the task.
+
+        :param task_instance: The task instance to handle the submit event for.
+        :param session: The session to be used for the database callback sink.
+        """
+        next_kwargs = task_instance.next_kwargs or {}
+        next_kwargs["event"] = self.payload
+        task_instance.next_kwargs = next_kwargs
+        task_instance.trigger_id = None
+        task_instance.state = TaskInstanceState.SCHEDULED
+
+
+class BaseTaskEndEvent(TriggerEvent):
+    """Base event class to end the task without resuming on worker."""
```

Comment:
Should we add `:meta private:` here? I feel this is probably something we don't need to make it public

Reply:
Added